### PR TITLE
Remove stack traces from web APIs

### LINKED ISSUE #2269: Security: stack traces returned in API
While gravity is OSS software, so returning stack traces doesn't tend to be a real information disclosure, it does provide some opportunity to observe the code paths or possibly the version of gravity in use. These aren't used, so we should disable them.

Example:
```
{
"error": {
"message": "access denied"
},
"traces": [
{
"path": "/gopath/src/github.com/gravitational/gravity/vendor/github.c
om/gravitational/teleport/lib/httplib/httplib.go",
"func": "github.com/gravitational/gravity/vendor/github.com/gravitati
onal/teleport/lib/httplib.WithCSRFProtection.func1",
"line": 85
},
{
"path": "/gopath/src/github.com/gravitational/gravity/vendor/github.c
om/julienschmidt/httprouter/router.go",
"func": "github.com/gravitational/gravity/vendor/github.com/juliensch
midt/httprouter.(*Router).ServeHTTP",
"line": 299
},
```

### PR DESCRIPTION
**Pull request by lenko-d**
## Description

This PR unwraps errors in web handlers in order to remove stack traces from web API calls.

## Type of change

* Bug fix (non-breaking change which fixes an issue)
* This change has a user-facing impact

## Linked tickets and other PRs

* Closes [2269](https://github.com/gravitational/gravity/issues/2269)
* Requires [5070](https://github.com/gravitational/teleport/pull/5070)
* Requires  [60](https://github.com/gravitational/trace/pull/60)

## TODOs

- [X] Self-review the change
- [X] Perform manual testing
- [X] Address review feedback
- [X] Update upstream references / tags / versions after upstream PR merges (linked above)

### Verification of the Gravity web portal GUI
Changed the code to trigger exceptions at several places and verified that the GUI still shows the error correctly without any issues.

### Verification of the APIs
Verified that the following APIs don't return the stacktraces:
```
curl -k -H "Content-Type: application/json" -d '{"username": "example", "password":"invalid"}' https://192.168.1.102:32009/portalapi/v1/sites
{
    "error": {
        "message": "unauthorized"
    }
}
```

```
 curl -k -H "Content-Type: application/json" -d '{"username": "example", "password":"invalid"}' https://192.168.1.102:32009/portalapi/v1/sites/sites/domain/users/username/reset
{
    "error": {
        "message": "method not found"
    }
}
```

```
curl -k -H "Content-Type: application/json"  https://192.168.1.102:32009/portalapi/v1/tokens/user/323223
{
    "error": {
        "message": "bad or expired token"
    }
}
```

```
curl -k -H "Content-Type: application/json" -d '{"username": "example", "password":"invalid"}' https://192.168.1.102:32009/proxy/v1/webapi/sessions/renew
{
    "error": {
        "message": "missing session cookie"
    }
}
```

### CODE REVIEW

**Review by a-palchikov — APPROVED** [review]

File: lib/ops/opshandler/opshandler.go [hunk 1]
```diff
@@ -358,7 +358,8 @@ func (h *WebHandler) getSiteInstructions(w http.ResponseWriter, r *http.Request,
 	serverProfile := p[1].Value
 	instructions, err := h.cfg.Operator.GetSiteInstructions(token, serverProfile, r.URL.Query())
 	if err != nil {
-		trace.WriteError(w, err)
+		log.WithError(err).Info(err.Error())
```
Comment (a-palchikov) [inline lib/ops/opshandler/opshandler.go:361]: nit: Specify the context for the error:
```go
log.WithError(err).Warn("Failed to query cluster install instructions.")
```

File: lib/blob/handler/blobhandler.go [hunk 2]
```diff
@@ -172,16 +172,17 @@ func (s *Server) needsAuth(fn authHandle, objects blob.Objects) httprouter.Handl
...
-			trace.WriteError(w, err)
+			log.WithError(err).Info(err.Error())
+			trace.WriteError(w, trace.Unwrap(err))
 			return
 		}
 
 		user, checker, err := s.cfg.Users.AuthenticateUser(*authCreds)
 		if err != nil {
-			log.Infof("authenticate error: %v", err)
+			log.WithError(err).Info("authenticate error")
```
Comment (a-palchikov) [inline lib/blob/handler/blobhandler.go:182]: nit: keep the error message consistent:
```go
log.WithError(err).Info("Authentication error.")

```

File: lib/blob/handler/blobhandler.go [hunk 3]
```diff
@@ -172,16 +172,17 @@ func (s *Server) needsAuth(fn authHandle, objects blob.Objects) httprouter.Handl
 
 		authCreds, err := httplib.ParseAuthHeaders(r)
 		if err != nil {
-			trace.WriteError(w, err)
+			log.WithError(err).Info(err.Error())
```
Comment (a-palchikov) [inline lib/blob/handler/blobhandler.go:175]: nit: add context to the message:
```go
log.WithError(err).Warn("Invalid authz headers.")
```

File: lib/blob/handler/blobhandler.go [hunk 4]
```diff
@@ -88,8 +88,8 @@ func New(cfg Config) (*Server, error) {
 
 func (s *Server) notFound(w http.ResponseWriter, r *http.Request) {
 	err := trace.NotFound("%v %v is not recognized", r.Method, r.URL.String())
-	log.Infof(err.Error())
-	trace.WriteError(w, err)
+	log.WithError(err).Info(err.Error())
```
Comment (a-palchikov) [inline lib/blob/handler/blobhandler.go:91]: nit: add context to the error:
```go
log.WithFields(logrus.Fields{
	logrus.ErrorKey: err,
	"method": r,Method,
	"url": r.URL.String(),
}).Warn("Invalid request.")
```

**Review by a-palchikov — APPROVED** [review]

**Review by knisbet — APPROVED** [review]

**Pull request merged**